File: Sorts & Search & Shuffling Algorithms/Union Find/QuickFind_UF.cpp
#include <iostream>
#include <vector>
using namespace std;

class QuickFindUF {
public:
    int size;
    vector<int> id;

    QuickFindUF(int N) {
        id = vector<int>(N);
        size = N;
        for (int i = 0; i < N; i++)
            id[i] = i;
    }

    bool connected(int p, int q) { return id[p] == id[q]; }

    void Union(
    int p,
    int q
    ){
        int pid = id[p];
        int qid = id[q];
        for (int i = 0; i < size; i++)
            if (id[i] == pid) id[i] = qid;
    }
};

File: Sorts & Search & Shuffling Algorithms/Union Find/QuickFind_UF.py
class QuickFind_UF:
    def __init__(self, size):
        self.id = [i for i in range(0, size)]
        self.size = size

    def connected(self, p, q):
        return self.id[p] == self.id[q]

    def union(self, p, q):
        pid = self.id[p]
        qid = self.id[q]
        for i in range(0, self.size):
            if self.id[i] == pid:
                self.id[i] = qid

    def count(self):
        return self.size
